### ISSUE TITLE
Conda unable to find packages that definitely exist - PackagesNotFoundError

### ISSUE THREAD

**Issue by sean-workman** [conda/conda#13550]
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

I am trying to install a conda package from the bioconda channel, specifically pysam. 

I have tried installing the package in an existing environment, and it will not work:

```
conda install bioconda::pysam
Retrieving notices: ...working... done
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - bioconda::pysam

Current channels:

  - https://conda.anaconda.org/bioconda/osx-arm64
  - https://conda.anaconda.org/bioconda/noarch
  - https://conda.anaconda.org/conda-forge/osx-arm64
  - https://conda.anaconda.org/conda-forge/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

I have also tried creating a new environment containing only the package and I am met with the same error:

```
conda create -n pysam bioconda::pysam
Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed

PackagesNotFoundError: The following packages are not available from current channels:

  - bioconda::pysam

Current channels:

  - https://conda.anaconda.org/bioconda/osx-arm64
  - https://conda.anaconda.org/bioconda/noarch
  - https://conda.anaconda.org/conda-forge/osx-arm64
  - https://conda.anaconda.org/conda-forge/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

This happened with another package recently, though I can't remember specifically what I was trying to install.

### Conda Info

```shell
active environment : base
    active env location : /opt/miniforge3
            shell level : 1
       user config file : /Users/sean/.condarc
 populated config files : /opt/miniforge3/.condarc
                          /Users/sean/.condarc
          conda version : 23.3.1
    conda-build version : not installed
         python version : 3.11.5.final.0
       virtual packages : __archspec=1=arm64
                          __osx=13.5=0
                          __unix=0=0
       base environment : /opt/miniforge3  (writable)
      conda av data dir : /opt/miniforge3/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/bioconda/osx-arm64
                          https://conda.anaconda.org/bioconda/noarch
                          https://conda.anaconda.org/conda-forge/osx-arm64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /opt/miniforge3/pkgs
                          /Users/sean/.conda/pkgs
       envs directories : /opt/miniforge3/envs
                          /Users/sean/.conda/envs
               platform : osx-arm64
             user-agent : conda/23.3.1 requests/2.31.0 CPython/3.11.5 Darwin/22.6.0 OSX/13.5
                UID:GID : 501:20
             netrc file : None
           offline mode : False
```

### Conda Config

```shell
==> /opt/miniforge3/.condarc <==
channels:
  - conda-forge

==> /Users/sean/.condarc <==
channels:
  - bioconda
  - conda-forge
offline: False
```

### Conda list

```shell
# packages in environment at /opt/miniforge3:
#
# Name                    Version                   Build  Channel
boltons                   23.1.1             pyhd8ed1ab_0    conda-forge
brotli-python             1.1.0           py311ha891d26_1    conda-forge
bzip2                     1.0.8                h93a5062_5    conda-forge
c-ares                    1.26.0               h93a5062_0    conda-forge
ca-certificates           2023.11.17           hf0a4a13_0    conda-forge
certifi                   2023.11.17         pyhd8ed1ab_0    conda-forge
cffi                      1.16.0          py311h4a08483_0    conda-forge
charset-normalizer        3.3.2              pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
conda                     23.3.1          py311h267d04e_0    conda-forge
conda-libmamba-solver     23.3.0             pyhd8ed1ab_0    conda-forge
conda-package-handling    2.2.0              pyh38be061_0    conda-forge
conda-package-streaming   0.9.0              pyhd8ed1ab_0    conda-forge
cryptography              42.0.2          py311h08c85a6_0    conda-forge
fmt                       9.1.0                hffc8910_0    conda-forge
icu                       73.2                 hc8870d7_0    conda-forge
idna                      3.6                pyhd8ed1ab_0    conda-forge
jsonpatch                 1.33               pyhd8ed1ab_0    conda-forge
jsonpointer               2.4             py311h267d04e_3    conda-forge
krb5                      1.21.2               h92f50d5_0    conda-forge
libarchive                3.6.2                h82b9b87_1    conda-forge
libcurl                   8.5.0                h2d989ff_0    conda-forge
libcxx                    16.0.6               h4653b0c_0    conda-forge
libedit                   3.1.20191231         hc8eb9b7_2    conda-forge
libev                     4.33                 h93a5062_2    conda-forge
libexpat                  2.5.0                hb7217d7_1    conda-forge
libffi                    3.4.2                h3422bc3_5    conda-forge
libiconv                  1.17                 h0d3ecfb_2    conda-forge
libmamba                  1.4.2                h7d1d596_0    conda-forge
libmambapy                1.4.2           py311hf90d2e5_0    conda-forge
libnghttp2                1.58.0               ha4dd798_1    conda-forge
libsolv                   0.7.27               h9e231a4_0    conda-forge
libsqlite                 3.44.2               h091b4b1_0    conda-forge
libssh2                   1.11.0               h7a5bd25_0    conda-forge
libxml2                   2.12.4               h0d0cfa8_1    conda-forge
libzlib                   1.2.13               h53f4e23_5    conda-forge
lz4-c                     1.9.4                hb7217d7_0    conda-forge
lzo                       2.10              h642e427_1000    conda-forge
mamba                     1.4.2           py311hb045da1_0    conda-forge
ncurses                   6.4                  h463b476_2    conda-forge
openssl                   3.2.1                h0d3ecfb_0    conda-forge
packaging                 23.2               pyhd8ed1ab_0    conda-forge
pip                       23.2.1             pyhd8ed1ab_0    conda-forge
pluggy                    1.4.0              pyhd8ed1ab_0    conda-forge
pybind11-abi              4                    hd8ed1ab_3    conda-forge
pycosat                   0.6.6           py311heffc1b2_0    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pyopenssl                 24.0.0             pyhd8ed1ab_0    conda-forge
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
python                    3.11.5          h47c9636_0_cpython    conda-forge
python_abi                3.11                    4_cp311    conda-forge
readline                  8.2                  h92ec313_1    conda-forge
reproc                    14.2.4.post0         h93a5062_1    conda-forge
reproc-cpp                14.2.4.post0         h965bd2d_1    conda-forge
requests                  2.31.0             pyhd8ed1ab_0    conda-forge
ruamel.yaml               0.17.40         py311h05b510d_0    conda-forge
ruamel.yaml.clib          0.2.7           py311heffc1b2_2    conda-forge
setuptools                69.0.3             pyhd8ed1ab_0    conda-forge
tk                        8.6.13               h5083fa2_1    conda-forge
toolz                     0.12.1             pyhd8ed1ab_0    conda-forge
tqdm                      4.66.1             pyhd8ed1ab_0    conda-forge
tzdata                    2023d                h0c530f3_0    conda-forge
urllib3                   2.2.0              pyhd8ed1ab_0    conda-forge
wheel                     0.42.0             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h57fd34a_0    conda-forge
yaml-cpp                  0.7.0                h13dd4ca_3    conda-forge
zstandard                 0.22.0          py311h67b91a1_0    conda-forge
zstd                      1.5.5                h4f39d0f_0    conda-forge
```

**Comment by travishathaway** [comment]
Hi @sean-workman,

I think this is happening because your hardware architecture is `osx-arm64` and this package is only available for `osx-64` (i.e. 64-bit architecture).

If you update to the latest version of conda, `23.11.0`, you could create a new environment for this package and specify `osx-64` as the architecture:

```
conda create --name pysam --platform osx-64 --channel bioconda pysam
```

Please let us know if that works for you.

**Comment by sean-workman** [comment]
Hi Travis,

Unfortunately I've been having a heck of a time trying to get things updated. I tried updating from a fresh install and it still didn't want to work. 

I have tried:

```
conda update -n base conda
Collecting package metadata (current_repodata.json): done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 23.3.1
  latest version: 23.11.0

Please update conda by running

    $ conda update -n base -c conda-forge conda

Or to minimize the number of packages updated during conda update use

     conda install conda=23.11.0



# All requested packages already installed.
```

Trying with
```
conda update -n base -c conda-forge conda
```

yields the same thing.

And if I try to specify the specific version:

```
conda install conda=23.11.0
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: \
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort
[...]
```

Followed by a massive output of conflicts, large enough that my terminal history doesn't allow me to scroll up and see it all.

**Comment by sean-workman** [comment]
Hi again,

I did another complete wipe of my install and started fresh. I had been using miniforge before and wonder if my insistence on trying to have things installed in /opt/ contributed to my problems because the fresh install in my home directory allowed me to get the most recent version of conda installed via the miniconda3 command line installer. 

Your solution of specifying an environment worked a treat. A little bit odd because I definitely didn't need to do that the last time I installed it, but I'm not going to fight it - it works!

Thanks again!
